projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-check-your-answers/query-check-your-answers.component.ts

```diff
@@ -304,6 +314,53 @@ export class QueryCheckYourAnswersComponent implements OnInit, OnDestroy {
     }
   }
 
+  private isNewQueryContext(data: CaseField): boolean {
+    return this.queryCreateContext === QueryCreateContext.NEW_QUERY && data.display_context !== this.DISPLAY_CONTEXT_READONLY;
+  }
+
+  private handleMultipleCollections(data: CaseField, id: string): boolean {
```

Comment:
I don't quite get this. I would expect to see some code where it gets a set of candidate collections, and selects the one with the lowest order, however only a single case field is passed into this function. I think it works by accumulating a value in this.fieldId, but that's a bit nasty. I'd rather you refactored this into a single function that fishes out the appropriate case field based on the ordering